lxd/device/cdi/configure.go

```diff
@@ -0,0 +1,292 @@
+package cdi
+
+import (
+	"fmt"
+	"os"
+	"path/filepath"
+	"strings"
+
+	"tags.cncf.io/container-device-interface/specs-go"
+
+	"github.com/canonical/lxd/lxd/instance"
+	"github.com/canonical/lxd/shared"
+	"github.com/canonical/lxd/shared/logger"
+)
+
+// specDevToNativeDev builds a list of unix-char devices to be created from a CDI spec.
+func specDevToNativeDev(configDevices *ConfigDevices, d *specs.DeviceNode) error {
+	if d == nil {
+		return fmt.Errorf("CDI device node is nil")
+	}
+
+	if d.HostPath == "" && d.Path == "" {
+		return fmt.Errorf("Both hostPath and path are empty in the CDI device node: %v", *d)
+	}
+
+	var devPath string
+	var hostPath string
+	if d.HostPath != "" && d.Path == "" {
+		devPath = d.HostPath
+		hostPath = d.HostPath
+	} else if d.HostPath == "" && d.Path != "" {
+		// When the hostPath is empty, the path is the device path in the container.
+		devPath = d.Path
+		hostPath = d.Path
+	} else {
+		devPath = d.Path
+		hostPath = d.HostPath
+	}
+
+	// choose a meaningful device name based on the devPath
+	deviceName := strings.Join(strings.Split(devPath, "/"), "__")
```

Comment:
why is this needed? Seems not appropriate to handle escaping in this package.

Reply:
do we even need a name?

Reply:
do we need a name at all? Can we just use a list of slices?

Reply:
I removed the need for names. Fixed.